File: src/main/java/guru/xgm/jass/openapi/fileTypes/JassFileType.java
package guru.xgm.jass.openapi.fileTypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import guru.xgm.jass.icons.JassIcons;
import guru.xgm.jass.lang.JassLanguage;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class JassFileType extends LanguageFileType {

    public static final JassFileType INSTANCE = new JassFileType();

    public JassFileType() {
        super(JassLanguage.INSTANCE);
    }

    public static final String extension = "j";

    @NotNull
    @Override
    public String getName() {
        return "JASS";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "JASS language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return extension;
    }

    @Override
    public @NotNull Icon getIcon() {
        return JassIcons.FILE;
    }
}

File: src/main/java/guru/xgm/language/jass/openapi/fileTypes/JassFileType.kt
package guru.xgm.language.jass.openapi.fileTypes

import com.intellij.openapi.fileTypes.LanguageFileType
import guru.xgm.language.jass.icons.JassIcons
import guru.xgm.language.jass.lang.JassLanguage
import javax.swing.Icon

// https://plugins.jetbrains.com/docs/intellij/language-and-filetype.html

class JassFileType : LanguageFileType(JassLanguage.instance) {
    override fun getName(): String = JassLanguage.NAME

    override fun getDescription(): String = JassLanguage.NAME

    override fun getDefaultExtension(): String = EXTENSION

    override fun getIcon(): Icon = JassIcons.FILE

    @Suppress("CompanionObjectInExtension")
    companion object {
        val instance: JassFileType = JassFileType()

        const val EXTENSION: String = "j"
    }
}
